committeeoversightapp/management/commands/import_data.py

```diff
@@ -115,98 +109,165 @@ def add_house_hearings(self):
-                        try:
-                            hearing_category = HearingCategory.objects.create(event=event, category_id=category)
-                        except IntegrityError:
-                            self.bad_rows.append("Row " + str(row_count) + ": Unrecognized category " + category + " on " + event.name)
+                        except AttributeError:
+                            self.bad_rows.append("Row " + str(self.row_count) + ": Unrecognized hearing number " + hearing_number_raw + " on " + name)
 
-                    row_count += 1
+                    elif name:
+                        self.smart_match(matched, name, committees_filtered, start_date, category, source, classification)
 
-    def get_committee(self, committee_name, parent):
+    def get_committee(self, committee_name, parent, committee_key):
         classification = "committee"
-        organization = Organization.objects.filter(Q(name__icontains=committee_name, parent=parent, classification=classification) |
-                    Q(other_names__name__icontains=committee_name, parent=parent, classification=classification))
-        if organization.count() == 1:
-            return organization[0]
-        elif organization.count() == 0:
-            return IndexError
+        organization_exact = Organization.objects.filter(name=committee_name, parent=parent, classification=classification)
+        organization_contains = Organization.objects.filter(name__icontains=committee_name, parent=parent, classification=classification)
+
+        if organization_exact.count() == 1:
+            return organization_exact[0]
         else:
-            return ValueError
+            if organization_contains.count() == 1:
+                return organization_contains[0]
+            elif organization_contains.count() == 0:
+                organization_other_name = Organization.objects.filter(other_names__name__icontains=committee_name, parent=parent, classification=classification)
+                try:
+                    return organization_other_name[0]
+                except IndexError:
+                    new_committee, _ = Committee.objects.get_or_create(lugar_id=committee_key, lugar_name=committee_name)
+                    self.bad_rows.append("Not recognized: " + committee_key + ", " + committee_name)
+            else:
+                self.bad_rows.append("Multiple possible committees for " + committee_key + ": " + committee_name)
 
-    def new_committee(self, committee_key, event, row_count):
+    def new_event_participant(self, committee_key, event):
         try:
             name = Committee.objects.get(lugar_id=committee_key).organization.name
             organization = Committee.objects.get(lugar_id=committee_key).organization
             entity_type = "organization"
             committee, created = EventParticipant.objects.get_or_create(name=name, event=event, organization=organization, entity_type=entity_type)
```

Comment:
pretty sure tthis line can just be 

EventParticipant.objects.get_or_create(event=event, organization=organization)

Reply:
Elsewhere these are saved with name and entity type attached (there are other event participants that aren't organizations, like witnesses). Why do you suggest not including that here?

Reply:
Ah, I see I'm doing redundant work to get the name. Fixed that, but I think I need to save it with both name and entity type information. Could you say more?